scripts/run_e2e_tests.py

```diff
@@ -596,7 +596,8 @@ def main(args=None):
         output, return_code = run_tests(parsed_args)
         # Don't rerun passing tests.
         if return_code == 0:
-            flake_checker.report_pass(parsed_args.suite)
+            if flake_checker.check_if_on_ci():
```

Comment:
[optional] Maybe:
```
# Don't rerun tests on CI.
if return_code == 0 and  if flake_checker.check_if_on_ci():
    ....
```
(To avoid multi-level if blocks)

Reply:
Here's the code there:

```python
        if return_code == 0:
            if flake_checker.check_if_on_ci():
                flake_checker.report_pass(parsed_args.suite)
            break
```

Notice that because of the `break` at the end, I think your proposed solution would make the code overly complicated. We would have to do:

```python
        if return_code == 0 and flake_checker.check_if_on_ci():
            flake_checker.report_pass(parsed_args.suite)
            break
        elif return_code == 0:
            break
```

Reply:
We would have to break the `for loop` irrespective of return code for `not  flake_checker.check_if_on_ci()`, right? If so, should we do it this way:

```
if not flake_checker.check_if_on_ci():
     break

if return_code == 0:
     flake_checker.report_pass(parsed_args.suite) 
     break
...
```

Reply:
done